go/ql/src/Security/CWE-798/HardcodedCredentials.ql

```diff
@@ -31,22 +32,35 @@ predicate isSensitive(DataFlow::Node sink, SensitiveExpr::Classification type) {
   )
 }
 
-from DataFlow::Node source, string message, DataFlow::Node sink, SensitiveExpr::Classification type
-where
+predicate sensitiveAssignment(
+  DataFlow::Node source, DataFlow::Node sink, SensitiveExpr::Classification type
+) {
   exists(string val | val = source.getStringValue() and val != "" |
-    isSensitive(sink, type) and
     DataFlow::localFlow(source, sink) and
+    isSensitive(sink, type) and
     // allow obvious dummy/test values
     not PasswordHeuristics::isDummyPassword(val) and
     not sink.asExpr().(Ident).getName().regexpMatch(HeuristicNames::notSensitive())
-  ) and
+  )
+}
+
+predicate hardcodedPrivateKey(DataFlow::Node node, SensitiveExpr::Classification type) {
+  node.getStringValue()
+      .regexpMatch("(?s)-+BEGIN\\b.*\\bPRIVATE KEY-+.+-+END\\b.*\\bPRIVATE KEY-+\n?") and
+  (node.asExpr() instanceof StringLit or node.asExpr() instanceof AddExpr) and
```

Comment:
Is there anything with a `getStringValue` and that matches that regex we wouldn't want to consider?

Reply:
Do you mean that `(node.asExpr() instanceof StringLit or node.asExpr() instanceof AddExpr)` is not necessary? If so, yes, you're probably right, `getStringValue` already restricts it enough.

```suggestion
```